From 6dc9c9e8b0b51abd9a332f5f4767df729848d579 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Fri, 19 Dec 2008 17:21:40 -0800
Subject: [PATCH 04/14] Staging: add android framework

This prepares us to start adding the android drivers
to the build.

The dummy android.c file will go away in the next few patches, as it
will not be needed once drivers/staging/android/ has a driver in it.

Cc: Robert Love <rlove@google.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/Kconfig           |    2 ++
 drivers/staging/Makefile          |    1 +
 drivers/staging/android/Kconfig   |    9 +++++++++
 drivers/staging/android/Makefile  |    1 +
 drivers/staging/android/android.c |   18 ++++++++++++++++++
 5 files changed, 31 insertions(+), 0 deletions(-)
 create mode 100644 drivers/staging/android/Kconfig
 create mode 100644 drivers/staging/android/Makefile
 create mode 100644 drivers/staging/android/android.c

diff --git a/drivers/staging/Kconfig b/drivers/staging/Kconfig
index c063971..ce6badd 100644
--- a/drivers/staging/Kconfig
+++ b/drivers/staging/Kconfig
@@ -93,5 +93,7 @@ source "drivers/staging/frontier/Kconfig"
 
 source "drivers/staging/epl/Kconfig"
 
+source "drivers/staging/android/Kconfig"
+
 endif # !STAGING_EXCLUDE_BUILD
 endif # STAGING
diff --git a/drivers/staging/Makefile b/drivers/staging/Makefile
index 53478e8..9ddcc2b 100644
--- a/drivers/staging/Makefile
+++ b/drivers/staging/Makefile
@@ -29,3 +29,4 @@ obj-$(CONFIG_USB_RSPI)		+= rspiusb/
 obj-$(CONFIG_INPUT_MIMIO)	+= mimio/
 obj-$(CONFIG_TRANZPORT)		+= frontier/
 obj-$(CONFIG_EPL)		+= epl/
+obj-$(CONFIG_ANDROID)		+= android/
diff --git a/drivers/staging/android/Kconfig b/drivers/staging/android/Kconfig
new file mode 100644
index 0000000..229f8d7
--- /dev/null
+++ b/drivers/staging/android/Kconfig
@@ -0,0 +1,9 @@
+menu "Android"
+
+config ANDROID
+	bool "Android Drivers"
+	default N
+	---help---
+	  Enable support for various drivers needed on the Android platform
+
+endmenu
diff --git a/drivers/staging/android/Makefile b/drivers/staging/android/Makefile
new file mode 100644
index 0000000..a551e53
--- /dev/null
+++ b/drivers/staging/android/Makefile
@@ -0,0 +1 @@
+obj-$(CONFIG_ANDROID)			+= android.o
diff --git a/drivers/staging/android/android.c b/drivers/staging/android/android.c
new file mode 100644
index 0000000..b8ad5f3
--- /dev/null
+++ b/drivers/staging/android/android.c
@@ -0,0 +1,18 @@
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/module.h>
+
+static int __init android_init(void)
+{
+	return 0;
+}
+
+static void __exit android_exit(void)
+{
+}
+
+module_init(android_init);
+module_exit(android_exit);
+
+MODULE_AUTHOR("Greg Kroah-Hartman");
+MODULE_LICENSE("GPL");
-- 
1.7.0.4

